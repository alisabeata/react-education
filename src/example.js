export const sum = (a, b) => a + b;

export const pushToArray = (arr, item) => [...arr, item];

export const promiseTimeout = (fn, timeout) => 
  new Promise(resolve => {
    setTimeout(() => {
      resolve(fn());
    }, timeout);
  });

export const multipl = (a, b) => a * b;