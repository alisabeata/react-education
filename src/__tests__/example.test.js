import {sum, pushToArray, promiseTimeout, multipl} from '../example';


describe('example.js', () => {
  it('sum is work', () => {
    expect(sum(1, 2)).toEqual(3);
  });
  
  it('pushToArray return new array' , () => {
    const ar = [1, 2, 3];
  
    expect(pushToArray(ar, 4)).toEqual([1, 2, 3, 4]);
  });

  it('promiseTimeout return promise after timeout' , done => {
    expect(
      promiseTimeout(() => {
        done();
      }, 1000)
    ).toBeInstanceOf(Promise);
  });

  it('multipl' , () => {
    expect(multipl(100, 100)).toMatchSnapshot();
  });
});

