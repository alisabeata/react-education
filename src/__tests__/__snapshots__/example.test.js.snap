// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example.js multipl 1`] = `10000`;
